### PR TITLE
Checkout: Make checkout button 'busy' when processing

### PR DESCRIPTION
**Pull request by aidvu**
Add feedback to checkout button. It would just go disabled and have no feedback that something is happening.

Before:
checkout button old

After:
checkout button

### CODE REVIEW

File: client/my-sites/checkout/checkout/pay-button.jsx [hunk 1]
```diff
@@ -142,17 +141,18 @@ class PayButton extends React.Component {
...
+		const buttonState = this.buttonState();
 
 		return (
 			<span className="pay-button">
-				<button
+				<Button
 					type="submit"
 					className="button is-primary button-pay pay-button__button"
+					busy={ buttonState.disabled }
 					disabled={ buttonState.disabled }
```
Comment (delputnam) [inline client/my-sites/checkout/checkout/pay-button.jsx]: I don't think we need this any longer. (Actually, I don't think this prop is even used for `Button`.)

**Comment by danhauk** [comment]
I would prefer to do something more consistent with the regular payment processing when paying with credit card. We add a progress bar rather than a busy state to the button ([see here](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout/credit-card-payment-box.jsx#L118)). Is that something we could easily port to the credits payment box?

**Comment by aidvu** [comment]
I'll try that @danhauk. 👍

File: client/my-sites/checkout/checkout/pay-button.jsx (hunk 1 above)
Comment (aidvu) [reply client/my-sites/checkout/checkout/pay-button.jsx]: It's passed down to the `button` component.

**Comment by klimeryk** [comment]
@aidvu, just wandering what's up with this PR, since there hasn't been an update for 13 days (not that I'm better with my own tasks...).

**Comment by aidvu** [comment]
@klimeryk compared to this diff, refactoring it to be able to use the progress bar is going to take a bit more time. :) It's in my queue, but low prio.

**Comment by danhauk** [comment]
@aidvu Go for it. 👍

**Pull request merged**